Add "confused artist" variant

fbarbe00 opened the pull request:
This variant gives to one person a different word from the same category, without the person knowing whether they have the same person as other players or the different one.

fbarbe00 committed:
Add confused artist variant

fbarbe00 committed:
remove stats link

fbarbe00 committed:
Categories are stored between games

fbarbe00 committed:
Fix explanation

fbarbe00 committed:
Capitalise categories in spanish

alenros commented:
Sorry for taking so long to review this!
I'm not sure I understand this variant: why does the person that gets the confused role have a different icon? doesn't it mean he will know he is confused?

fbarbe00 commented:
No worries! Yeah I guess it's not too clear the way I wrote it... But basically everyone gets the same icon (which is just a question mark - I tested this), so no one knows whether they have the same word as the others or not. One person will have a different word from the same category, and needs to realise they have a different word and blend in. Because no one is sure of their word, people will be a lot more cautious and vague drawing their word, resulting in a much more abstract drawing.

Note that I came up with this on my own and would have no problem having this variant rejected.

alenros commented:
It's very similar to a variant I thought of: #45, The Mad Artist: one of the real artists becomes mad, getting a different word from the same category as everyone else. he thinks he is a real artist, but he can only win by voting for himself (ie signaling he understood he is mad).

Not sure if mine is better, but I think one of these is enough.

let's discuss and pick one. feel free to be impolite and pick my variant apart or to simply prefer your own, and then I will merge it.

fbarbe00 commented:
Ah yes indeed, this seems very similar to what you came up with! The only difference, with my understanding, is that in your described mad artist variant, there would still be a fake artist. And as you described I also wouldn't know how the winning condition could be for that one. Would the real artists need to spot the mad artist? Does the mad artist need to guess the word? Do players get two votes?

This confused artist variant seems to be much easier to play, so I think I prefer this one. I have played a few rounds of this (with only three people), and I do have to say that spotting the confused artist was quite easy. But the variant was easy to understand.

alenros commented:
The premise for Mad Artist is that there is still a FA, but he is harder to spot because the Mad Artist misleads the Real Artists.
The variant is intended for groups where spotting the FA got too easy.

The players still vote only once for finding the Fake Artist.
Win conditions:
1. The Real Artists: as in a regular game, but it's harder since the Mad Artist also draws off-target.
2. The Fake Artist: as in a regular game, but it's easier since the Mad Artist is an easy scape goat, and also allows the FA to claim he thinks he is a Mad Artist and that's why his drawings doesn't match.
3. The Mad Artist: has a hard time figuring who the FA is since he needs to figure out what's been drawn himself, and then understand that he's not in sync with the group. to win he must show he understood he's a Mad Artist and so needs to vote for himself. this would be a losing move if you are a Real Artist since the FA is more likely to not be voted.


Having explained all this, I think the two variants achieve different things, and maybe we can have both.

fbarbe00 committed:
Major rewrite: port to latest node version, make dockerfile lighter

This commit focuses on making the docker image lighter, and updates dependencies, removing outdated ones.
